src/tools/roc_conv/main.cpp

```diff
@@ -68,23 +69,33 @@ int main(int argc, char** argv) {
 
     pipeline::ConverterConfig config;
 
-    if (args.frame_size_given) {
-        if (args.frame_size_arg <= 0) {
-            roc_log(LogError, "invalid --frame-size: should be > 0");
+    if (args.frame_length_given) {
+        if (!core::parse_duration(args.frame_length_arg, config.internal_frame_length)) {
+            roc_log(LogError, "invalid --frame-length: frame size unspecified");
             return 1;
         }
-        config.internal_frame_size = (size_t)args.frame_size_arg;
+        if (packet::ns_to_size(config.internal_frame_length, config.input_sample_rate,
+                               config.input_channels) <= 0) {
+            roc_log(LogError, "invalid --frame-length: should be > 0");
+            return 1; 
+        }
     }
 
-    sndio::BackendDispatcher::instance().set_frame_size(config.internal_frame_size);
+    sndio::BackendDispatcher::instance().set_frame_size(
+        packet::ns_to_size(config.internal_frame_length, config.input_sample_rate,
+                           config.input_channels));
 
-    core::BufferPool<audio::sample_t> pool(allocator, config.internal_frame_size,
-                                           args.poisoning_flag);
+    core::BufferPool<audio::sample_t> pool(
+        allocator, packet::ns_to_size(config.internal_frame_length,
+                                      config.input_sample_rate, config.input_channels),
+        args.poisoning_flag);
 
     sndio::Config source_config;
     source_config.channels = config.input_channels;
     source_config.sample_rate = 0;
-    source_config.frame_size = config.internal_frame_size;
+    source_config.frame_size = packet::ns_to_size(config.internal_frame_length,
```

Comment:
We should also replace sndio::Config::frame_size (in bytes) with sndio::Config::frame_length (in nanoseconds) and perform the conversion inside sndio.